### DISCUSSION TITLE
Error: Must use import to load ES Module: D:\node_modules\react-markdown\index.js require() of ES modules is not supported

### DISCUSSION THREAD

**Discussion by Jay-Bruh** [remarkjs/remark#835] [Q&A] [answered]
Currently I'm using "react": "17.0.2" and I have installed "react-markdown": "^7.0.1" via npm i react-markdown I'm using this package to display my rich text that I'm fetching from my Strapi CMS. I have used the following code to display the content:

import ReactMarkdown from "react-markdown";
```
export default function name({ posts }) {
  return (
    <>
      <div>
        <div>
          {posts.Title}
        </div>
      </div>
      <div>
        <ReactMarkdown source={posts.Content} escapeHtml={false} />
      </div>
    </>
  );
}

export async function getStaticProps() {
  const res = await fetch("http://localhost:1337/blogs");
  const posts = await res.json();

  return {
    props: { posts },
  };
}
```

But when I run this it gives me the following error:

![image](https://user-images.githubusercontent.com/50700356/131652749-50148c0d-bb32-42be-b620-21f660f50e9e.png)

I'm using node v14.17.0 and have tried adding "type": "module".

My package.json:
```
{
  "name": "passportlegacy",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "axios": "^0.21.1",
    "babel-plugin-inline-react-svg": "^2.0.1",
    "next": "11.0.1",
    "next-images": "^1.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-map-gl": "^6.1.16",
    "react-markdown": "^7.0.1",
  },
  "devDependencies": {
    "@svgr/webpack": "^5.5.0",
    "@types/react": "17.0.16",
    "eslint": "7.32.0",
    "eslint-config-next": "11.0.1",
    "typescript": "4.3.5"
  }
}
```

**Comment by wooorm** [comment, marked as answer]
`react-markdown` and all of unified is ESM only. See https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c for more info on ESM and what your options are.

**Reply by Jay-Bruh** [reply]
Okay so I have tried with pure ESM now. I edit my content on my CMS and it looks like this:
![image](https://user-images.githubusercontent.com/50700356/131660690-dba05d3c-cee9-495a-bd38-06f35949b50d.png)

And now my current code is:
```
<main className={styles.main}>
        {posts &&
          posts.map((image) => (
            <div style={{ height: "100%" }}>
              <ReactMarkdown children={image.Content} skipHtml={false} />
            </div>
          ))}
      </main>

export async function getStaticProps() {
  const res = await fetch("http://localhost:1337/blogs");
  const posts = await res.json();

  return {
    props: { posts },
  };
}
```

This shows all the spaces that I have made in my webpage but this also brings up the tags on my webpage:

![image](https://user-images.githubusercontent.com/50700356/131660897-4dcca6a7-f7e2-46f9-bad5-3d951a11ca9c.png)

How do I make it so that it'll actually make my < u > tags an actual underline and not just put it as is in my webpage

**Reply by wooorm** [reply]
This is explained in the readme of `react-markdown`. Did you read that? If so, can you please clarify *what* information is missing or confusing there?

**Reply by Jay-Bruh** [reply]
I did try the attributes that were there in readme file, but I'm still getting the same output on my webpage screen for some reason

**Reply by wooorm** [reply]
See Appendix A: https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown

**Comment by phil-lgr** [comment]
I've spent close to 2 hours trying to convert my existing webpack setup to esm and simply put, it is not an easy task. I don't understand why package author don't publish libraries in CJS and ESM, this is very simple using the `exports` property now.

Has anyone been able to use this package with CJS webpack config files? Which is still the default for webpack https://webpack.js.org/configuration/

**Reply by ChristianMurphy** [reply]
> I don't understand why package author don't publish libraries in CJS and ESM, this is very simple using the exports property now.

It is simple but broken.
Dual packaging breaks many adopters and causes a host of issues.
See https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c?permalink_comment_id=3850849#gistcomment-3850849 for a list of some (but not all) issues which come up.

> Has anyone been able to use this package with CJS webpack config files? Which is still the default for webpack https://webpack.js.org/configuration/

I'm not sure what you're asking, or how it relates to this thread.
There are a number of existing webpack questions which may answer your question https://github.com/orgs/remarkjs/discussions?discussions_q=webpack

If they don't, please see the support guide on how to frame questions so others can effectively help https://github.com/remarkjs/.github/blob/main/support.md#asking-quality-questions and consider opening a new thread (https://github.com/orgs/remarkjs/discussions/new).
